axelor-base/src/main/resources/views/Partner.xml

```diff
@@ -379,19 +379,36 @@
-                placeholder="e.g. Financial department, CEO"/>
-              <field name="addressL3" x-bind="{{addressL3|uppercase}}"
-                placeholder="e.g. Building A, Entry 2, 2nd Floor, Door 2E"/>
-              <field name="addressL4" x-bind="{{addressL4|uppercase}}"/>
-              <field name="addressL5" x-bind="{{addressL5|uppercase}}"
-                placeholder="e.g. POB 254"/>
-              <field name="city" canNew="true" onChange="action-address-change-city"
-                form-view="city-form" grid-view="city-grid"/>
-              <field name="addressL6" x-bind="{{addressL6|uppercase}}" required="true"/>
-              <field name="addressL7Country" required="true" form-view="country-form"
-                grid-view="country-grid"/>
+            <editor x-viewer="true" x-show-titles="false"
+              onNew="action-group-base-address-country-change">
+              <panel name="mainAddressPanel" colSpan="12">
+                <field name="country" required="true" form-view="country-form"
+                  grid-view="country-grid" onChange="action-group-base-address-country-change"/>
+                <panel name="countryHiddenPanel" hideIf="country == null" colSpan="12">
+                  <field name="room" x-bind="{{room|uppercase}}"
+                    placeholder="e.g. Financial department, CEO"/>
+                  <field name="floor" x-bind="{{floor|uppercase}}"
+                    placeholder="e.g. Building A, Entry 2, 2nd Floor, Door 2E"/>
+                  <field name="street" onSelect="action-address-attrs-update-domain-street"
+                    onChange="action-address-record-change-streetName" form-view="street-form"
+                    grid-view="street-grid" canNew="true"
+                    if="__config__.app.getApp('base')?.getStoreStreets()"/>
+                  <field if="__config__.app.getApp('base')?.getStoreStreets()"
+                    name="streetName" x-bind="{{streetName|uppercase}}"/>
+                  <field name="buildingName" x-bind="{{buildingName|uppercase}}"/>
+                  <field name="department" x-bind="{{department|uppercase}}"/>
+                  <field name="subDepartment" x-bind="{{subDepartment|uppercase}}"/>
+                  <field name="townName" x-bind="{{townName|uppercase}}"/>
+                  <field name="townLocationName" x-bind="{{townLocationName|uppercase}}"/>
+                  <field name="districtName" x-bind="{{districtName|uppercase}}"/>
+                  <field name="countrySubDivision" x-bind="{{countrySubDivision|uppercase}}"/>
+                  <field name="postBox" x-bind="{{postBox|uppercase}}"
+                    placeholder="e.g. POB 254"/>
+                  <field name="city" canNew="true" onChange="action-address-change-city"
+                    onSelect="action-address-attrs-update-domain-city" form-view="city-form"
+                    grid-view="city-grid"/>
+                </panel>
+              </panel>
```

Comment:
Sorry I fought it was all the time like that for the address on a partner but it's only for the employee I see. 
So remove the editor, just show the field as a normal M2O